js_modules/dagster-ui/packages/ui-core/src/asset-graph/CollapsedGroupNode.tsx

```diff
@@ -121,6 +128,93 @@ const GroupNodeAssetStatusCounts = ({
   group,
 }: {
   group: GroupLayout & {assetCount: number; assets: GraphNode[]};
+}) => {
+  if (featureEnabled(FeatureFlag.flagUseNewObserveUIs)) {
+    return <GroupNodeAssetStatusCountsAssetHealth group={group} />;
+  }
+  return <GroupNodeAssetStatusCountsNonAssetHealth group={group} />;
+};
+
+const GroupNodeAssetStatusCountsAssetHealth = ({
+  group,
+}: {
+  group: GroupLayout & {assetCount: number; assets: GraphNode[]};
+}) => {
+  const assetKeys = React.useMemo(() => group.assets.map((node) => node.assetKey), [group.assets]);
+
+  const {liveDataByNode} = useAssetsHealthData(assetKeys, 'group-node');
+  const statuses = React.useMemo(() => {
+    return Object.values(liveDataByNode).reduce(
+      (acc, liveData) => {
+        const health = liveData.assetHealth?.assetHealth ?? AssetHealthStatus.HEALTHY;
+        acc[health] = acc[health] ?? [];
+        if (liveData.assetHealth) {
+          acc[health].push(liveData.assetHealth);
+        }
+        return acc;
+      },
+      {} as Record<AssetHealthStatus, NonNullable<AssetHealthFragment['assetHealth']>[]>,
+    );
+  }, [liveDataByNode]);
+
+  return (
+    <Box padding={{horizontal: 12, bottom: 4}} flex={{direction: 'row', gap: 4}}>
+      {Object.keys(liveDataByNode).length !== assetKeys.length ? (
+        <AssetDescription $color={Colors.textLighter()}>
+          {group.assetCount} {group.assetCount === 1 ? 'asset' : 'assets'} (fetching statuses)
+        </AssetDescription>
+      ) : (
+        <>
+          <>
+            {statuses[AssetHealthStatus.HEALTHY] ? (
+              <Tooltip
+                content={`${statuses[AssetHealthStatus.HEALTHY].length} ${ifPlural(
+                  statuses[AssetHealthStatus.HEALTHY].length,
+                  'asset is',
+                  'assets are',
+                )} healthy`}
+              >
+                <Tag
+                  icon={statusToIconAndColor[AssetHealthStatus.HEALTHY].iconName}
+                  intent="success"
+                >
+                  {numberFormatter.format(statuses[AssetHealthStatus.HEALTHY].length)}
+                </Tag>
+              </Tooltip>
+            ) : null}
+          </>
+          {statuses[AssetHealthStatus.WARNING] ? (
+            <Tooltip
+              content={`${statuses[AssetHealthStatus.WARNING].length} asset${ifPlural(
+                statuses[AssetHealthStatus.WARNING].length,
+                ' has',
+                's have',
```

Comment:
Would vaguely be nice to have `asset has`, `assets have` here similar to line 173-174, but tiny nit

Reply:
ah yeah I'll update it for consistency